sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -1869,11 +1870,21 @@ void ProgramManager::addImages(sycl_device_binaries DeviceBinary) {
 
     cacheKernelUsesAssertInfo(*Img);
 
-    // check if kernel uses asan
+    // check if kernel uses sanitizer
     {
-      sycl_device_binary_property Prop = Img->getProperty("asanUsed");
-      m_AsanFoundInImage |=
-          Prop && (detail::DeviceBinaryProperty(Prop).asUint32() != 0);
+      sycl_device_binary_property SanProp = Img->getProperty("sanUsed");
```

Comment:
By ignoring the old property you effectively break sanitizer support for applications which were built with older compiler, but ran using newer runtime.

In general, we do guarantee backwards compatibility, but I don't know about address sanitizers project status to say that the requirements are as strict as they are for core SYCL functionality (and non-experimental extensions).

Reply:
Sanitizer is used to help debug code, and it won't be included in product.
In general, User have to rebuild their code each time when they want to use sanitizer, so I think it has low impact on compatibility.

Reply:
If you insist, it's okay to me to add compatibility code for "asanUsed" property.